Assets/PatchKit Patcher/Scripts/Patcher.cs

```diff
@@ -580,6 +580,11 @@ private void EnsureSingleInstance()
                     _lockFileStream = File.Open(lockFilePath, FileMode.Append);
                     DebugLogger.Log("Lock file open success");
                 }
+                catch (UnauthorizedAccessException exception)
```

Comment:
Please check if this exception isn't actually thrown in multiple instances situation. If it is, an if statement checking for online/offline mode might be required here (as well as an appropriate comment explaining this mechanism).

Reply:
[output_log.txt](https://github.com/patchkit-net/patchkit-patcher-unity/files/5558852/output_log.txt)

These are logs from the launch of two patchers in online mode, you can see that one of them have an error:
`2020-11-18 09:47:33.867 +01:00 [ ERROR ] <Patcher::ThreadExecution> An exception has occured
PatchKit.Unity.Patcher.MultipleInstancesException: Another instance of Patcher spotted
  at PatchKit.Unity.Patcher.Patcher.EnsureSingleInstance () [0x00000] in <filename unknown>:0 
  at PatchKit.Unity.Patcher.Patcher.ThreadExecution (CancellationToken cancellationToken) [0x00000] in <filename unknown>:0 `

So UnauthorizedAccessException doesn't catch in this situation. I hope I understood your comment correctly.

Reply:
Did these patchers include your modification?

Reply:
Yes